include/fiction/algorithms/simulation/sidb/determine_electrostatic_influence_region.hpp

```diff
@@ -0,0 +1,285 @@
+     * It identifies cells in the layout that are not present in the sublayout and collects
+     * them.
+     *
+     * @tparam Lyt The layout type.
+     */
+    void collect_cells_of_layout_without_subcells() noexcept
+    {
+        std::vector<typename Lyt::cell> all_sidbs = {};
+        all_sidbs.reserve(layout.num_cells());
+        layout.foreach_cell([&all_sidbs](const auto& c1) { all_sidbs.push_back(c1); });
+
+        std::vector<typename Lyt::cell> all_sidbs_sublayout = {};
+        all_sidbs_sublayout.reserve(sublayout.num_cells());
+        sublayout.foreach_cell([&all_sidbs_sublayout](const auto& c1) { all_sidbs_sublayout.push_back(c1); });
+
+        for (const auto& cell : all_sidbs)
+        {
+            auto it = std::find(all_sidbs_sublayout.begin(), all_sidbs_sublayout.end(), cell);
+
+            if (it == all_sidbs_sublayout.end())
+            {
+                cells_without_sublayout_cells.insert(cell);
+            }
+        }
+    }
+    /**
+     * This function simulates the ground state of the sublayout using *QuickExact*.
+     * The function then collects the charge indices of all charge distributions being considered as ground states.
+     *
+     * @tparam Lyt The layout type.
+     */
+    void simulate_ground_state_charge_indices_of_sublayout() noexcept
+    {
+        const auto simulation_results = quickexact(sublayout, quickexact_parameter);
+        // find the ground state, which is the charge distribution with the lowest energy
+        const auto ground_state_energy = minimum_energy(simulation_results.charge_distributions);
+        for (const auto& lyt : simulation_results.charge_distributions)
+        {
+            if (std::fabs(lyt.get_system_energy() - ground_state_energy) < std::numeric_limits<double>::epsilon())
+            {
+                ground_state_charge_indices_of_sublyt.insert(lyt.get_charge_index_and_base().first);
+            }
+        }
+    }
+    /**
+     * It determines if every cell in the sublayout is part of the layout.
+     *
+     * @tparam Lyt The type of the layout.
+     * @return `true` if the sublayout is a subset of the layout, `false` otherwise.
+     */
+    [[nodiscard]] bool is_sublayout_of_layout() const noexcept
+    {
+        std::vector<typename Lyt::cell> all_sidbs = {};
+        all_sidbs.reserve(layout.num_cells());
+        layout.foreach_cell([&all_sidbs](const auto& c1) { all_sidbs.push_back(c1); });
+
+        std::vector<typename Lyt::cell> all_sidbs_sublayout = {};
+        all_sidbs_sublayout.reserve(layout.num_cells());
+        sublayout.foreach_cell([&all_sidbs_sublayout](const auto& c1) { all_sidbs_sublayout.push_back(c1); });
+
+        for (const auto& cell : all_sidbs_sublayout)
+        {
+            auto it = std::find(all_sidbs.begin(), all_sidbs.end(), cell);
+
+            if (it == all_sidbs.end())
+            {
+                return false;
+            }
+        }
+        return true;
```

Comment:
If you want to stay with `std::vector`s, this code is fine. You could explore using `std::set` or `phmap::btree_set` and apply `std::includes` for the same effect.

Reply:
Probably not used anymore anyways if you decide to look into `mockturtle::window_view`